CMake/CompilerFlags.cmake

```diff
@@ -12,7 +12,10 @@ if (WIN32)
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # Taken from https://github.com/BlueBrain/hpc-coding-conventions/blob/master/cpp/cmake/bob.cmake#L192-L255
   if(${PROJECT_NAME}_CXX_WARNINGS)
-    set(FLAGS "${FLAGS} -Werror -Weverything")
+    if(${PROJECT_NAME}_WERROR)
```

Comment:
I think it's good to be able to see warnings while developing. Often they're just that warnings, some are true some are false. Yes, one should work towards eliminating them all; otherwise one misses the important one.

My issue with `-Werror` is that it makes appeasing the compiler priority number one. Nothing else is possible until there are no more warnings.

Concretely, GCC 12.2.1 will emit many warnings from within third-party libraries.

Reply:
Ok, you've convinced me; we'll keep it.  Thanks.